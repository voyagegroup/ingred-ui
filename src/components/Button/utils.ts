import { Theme } from "../../themes";
import { ButtonColor, ButtonColorStyle } from "./Button";

export const getContainerColorStyles = (
  theme: Theme,
): { [P in ButtonColor]: ButtonColorStyle } => ({
  primary: {
    normal: {
      background: theme.palette.primary.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowPrimary"],
      border: `1px solid ${theme.palette.primary.dark}`,
    },
    hover: {
      background: theme.palette.primary.dark,
      border: "none",
    },
    active: {
      background: theme.palette.primary.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  basicLight: {
    normal: {
      background: theme.palette.background.default,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.divider}`,
    },
    hover: {
      background: theme.palette.gray.highlight,
      border: `1px solid ${theme.palette.divider}`,
    },
    active: {
      background: theme.palette.gray.highlight,
      boxShadow: theme.shadow["3dShadowActive"],
      border: `1px solid ${theme.palette.divider}`,
    },
  },
  basicDark: {
    normal: {
      background: theme.palette.basicDark.main,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.basicDark.deepDark}`,
    },
    hover: {
      background: theme.palette.basicDark.dark,
      border: "none",
    },
    active: {
      background: theme.palette.basicDark.dark,
      boxShadow: theme.shadow["3dShadowBasicActive"],
      border: "none",
    },
  },
  danger: {
    normal: {
      background: theme.palette.danger.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowDanger"],
      border: `1px solid ${theme.palette.danger.dark}`,
    },
    hover: {
      background: theme.palette.danger.dark,
      border: "none",
    },
    active: {
      background: theme.palette.danger.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  clear: {
    normal: {
      background: "none",
      color: theme.palette.black,
      boxShadow: "none",
      border: "none",
    },
    hover: {
      background: theme.palette.gray.light,
      border: "none",
    },
    active: {
      background: theme.palette.gray.main,
      boxShadow: "none",
      border: "none",
    },
  },
  primaryPale: {
    normal: {
      background: theme.palette.primaryPale.main,
      color: theme.palette.primaryPale.dark,
      boxShadow: theme.shadow["3dShadowPrimaryPale"],
      border: `1px solid ${theme.palette.primaryPale.light}`,
    },
    hover: {
      background: theme.palette.primaryPale.highlight,
      border: "none",
    },
    active: {
      background: theme.palette.primaryPale.highlight,
      boxShadow: theme.shadow["3dShadowPrimaryPaleActive"],
      border: "none",
    },
  },
});
