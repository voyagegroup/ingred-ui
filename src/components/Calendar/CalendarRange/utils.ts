import dayjs, { Dayjs } from "dayjs";

const isStart = (startDate: Dayjs | null, month: Dayjs, day: number) =>
  startDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isEnd = (endDate: Dayjs | null, month: Dayjs, day: number) =>
  endDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isBetween = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) =>
  (startDate &&
    endDate &&
    dayjs(new Date(month.year(), month.month(), day)).isAfter(
      startDate.format("YYYY-MM-DD"),
      "day",
    ) &&
    dayjs(new Date(month.year(), month.month(), day)).isBefore(
      endDate.format("YYYY-MM-DD"),
      "day",
    )) ??
  false;

export const getDayState = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) => {
  if (isStart(startDate, month, day)) {
    return "start";
  }
  if (isEnd(endDate, month, day)) {
    return "end";
  }
  if (isBetween(startDate, endDate, month, day)) {
    return "between";
  }
  return "none";
};
