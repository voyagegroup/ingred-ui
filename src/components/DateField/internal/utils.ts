export const getInputWidth = (input: string) =>
  [...input.split("")].reduce((acc, char) => {
    const code = char.charCodeAt(0);
    return acc + (code >= 0x3000 && code <= 0xff60 ? 2 : 1);
  }, 0);
