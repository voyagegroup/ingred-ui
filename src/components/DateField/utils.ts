import { useMemo } from "react";

type Sections = {
  start: number;
  end: number;
  value: string;
  editable: boolean;
};

/**
 * 何らかのフォーマットで入ってくる日付を開始位置と終了位置と値を持つセクションに分割する
 * useDateField で format された日付操作を汎用的に行うために必要なプロパティを返す
 * 例) 2023-01-02 -> [
 *   { start: 0, end: 3, value: "2023", editable: true },
 *   { start: 4, end: 4, value: "-", editable: false },
 *   { start: 5, end: 6, value: "01", editable: true },
 *   { start: 7, end: 7, value: "-", editable: false },
 *   { start: 8, end: 9, value: "02", editable: true }
 * ]
 *
 * @param formattedDate 何らかのフォーマットで入ってくる日付
 * @returns 開始位置と終了位置と値を持つセクション
 */
export const getSections = (formattedDate?: string | null) => {
  if (!formattedDate) {
    return [];
  }

  const sections: Sections[] = [];
  let start = 0;
  let isPrevCharDigit = !isNaN(Number(formattedDate[0]));

  for (let index = 1; index <= formattedDate.length; index++) {
    const currentChar = formattedDate[index];
    const isCurrentCharDigit =
      !isNaN(Number(currentChar)) && currentChar !== " ";

    if (
      isCurrentCharDigit !== isPrevCharDigit ||
      currentChar === " " ||
      index === formattedDate.length
    ) {
      sections.push({
        start,
        end: index - 1,
        value: formattedDate.slice(start, index),
        editable: isPrevCharDigit,
      });
      start = index;
      isPrevCharDigit = isCurrentCharDigit;
    }
  }

  return sections;
};

/**
 * 開始位置と終了位置と値を持つセクションをフォーマットされた日付に変換する
 */
export const formatString = (sectionsWithCharactor: Sections[]) =>
  sectionsWithCharactor.map((section) => section.value).join("");

type ReactRef<T> =
  | React.RefCallback<T>
  | React.MutableRefObject<T>
  | React.ForwardedRef<T>
  | string
  | null
  | undefined;

// from: https://github.com/voyagegroup/ingred-ui/blob/master/src/hooks/useMergeRefs.ts
export function useMergeRefs<T>(...refs: ReactRef<T>[]): React.Ref<T> {
  return useMemo(() => {
    if (refs.every((ref) => ref === null)) {
      return null;
    }
    return (refValue: T) => {
      for (const ref of refs) {
        if (typeof ref === "function") {
          ref(refValue);
        } else if (ref && typeof ref !== "string") {
          ref.current = refValue;
        }
      }
    };
  }, [refs]);
}
