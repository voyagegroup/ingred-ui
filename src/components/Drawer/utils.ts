/**
 * 相対値をpx値に変換するヘルパー関数
 * @param value - 変換する値（数値またはvw/vh/%/px等の単位付き文字列）
 * @param direction - Drawerの方向（変換時の基準値決定に使用）
 * @returns px値（数値）
 */
export const convertToPixels = (value: string | number, direction: "left" | "right" | "bottom"): number => {
  if (typeof value === "number") {
    return value; // 数値はpx扱い
  }
  
  // 文字列の場合は単位チェック
  const numericValue = parseFloat(value);
  
  if (value.includes("vw")) {
    return (numericValue / 100) * window.innerWidth;
  }
  if (value.includes("vh")) {
    return (numericValue / 100) * window.innerHeight;
  }
  if (value.includes("%")) {
    // %の場合は親要素基準だが、ここではviewport基準として扱う
    const isHorizontal = direction === "left" || direction === "right";
    return (numericValue / 100) * (isHorizontal ? window.innerWidth : window.innerHeight);
  }
  if (value.includes("px")) {
    return numericValue;
  }
  
  // 単位なしの文字列は警告を出す
  if (!isNaN(numericValue)) {
    console.warn(`Drawer: 文字列サイズには単位を指定してください。"${value}" → "${value}px" を推奨`);
    return numericValue; // 一応数値として扱う
  }
  
  // 無効な値の場合はエラー警告とフォールバック
  console.error(`Drawer: 無効なサイズ値です: "${value}". デフォルト値を使用します。`);
  return 400; // フォールバック
}; 