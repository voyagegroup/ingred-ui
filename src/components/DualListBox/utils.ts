import {
  CandidateItem,
  DualListBoxCandidateItem,
  DualListBoxItem,
} from "./DualListBox";

export const getCandidateItems = (
  candidateItems: DualListBoxCandidateItem[],
  selectedItems: DualListBoxItem[],
) => {
  const result: CandidateItem[] = [];
  candidateItems.forEach((candidateItem) => {
    if (candidateItem.items) {
      result.push({
        ...candidateItem,
        items: getCandidateItems(candidateItem.items, selectedItems),
      });
      return;
    }

    const targetItem = selectedItems.find(
      (selectedItem) => selectedItem.id === candidateItem.id,
    );
    if (targetItem) {
      result.push({
        ...candidateItem,
        selected: true,
      });
    } else {
      result.push(candidateItem);
    }
  });

  return result;
};
