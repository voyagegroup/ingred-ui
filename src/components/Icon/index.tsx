export { default } from "./Icon";
export type { Props as IconProps, IconName } from "./Icon";
