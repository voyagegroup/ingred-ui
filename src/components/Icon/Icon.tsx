import * as React from "react";
import * as Styled from "./styled";
import { Theme } from "../../themes";
import { useTheme } from "../../themes/useTheme";
import { DashboardIcon } from "./internal/DashboardIcon";
import { BarChartIcon } from "./internal/BarChartIcon";
import { LogoutIcon } from "./internal/LogoutIcon";
import { SettingIcon } from "./internal/SettingIcon";
import { ArrowBottomIcon } from "./internal/ArrowBottomIcon";
import { ArrowLeftIcon } from "./internal/ArrowLeftIcon";
import { ArrowRightIcon } from "./internal/ArrowRightIcon";
import { PencilIcon } from "./internal/PencilIcon";
import { DeleteBinIcon } from "./internal/DeleteBinIcon";
import { CloseIcon } from "./internal/CloseIcon";
import { ForbidIcon } from "./internal/ForbidIcon";
import { CheckIcon } from "./internal/CheckIcon";
import { EyeIcon } from "./internal/EyeIcon";
import { EyeOffIcon } from "./internal/EyeOffIcon";
import { ExclamationIcon } from "./internal/ExclamationIcon";
import { ArrowDoubleLeftIcon } from "./internal/ArrowDoubleLeftIcon";
import { QuestionIcon } from "./internal/QuestionIcon";
import { ZoomInIcon } from "./internal/ZoomInIcon";
import { SortUpIcon } from "./internal/SortUpIcon";
import { SortDownIcon } from "./internal/SortDownIcon";
import { SortInactiveIcon } from "./internal/SortInactiveIcon";
import { FolderIcon } from "./internal/FolderIcon";
import { FolderOpenIcon } from "./internal/FolderOpenIcon";
import { ReturnLineIcon } from "./internal/ReturnLineIcon";
import { LinkIcon } from "./internal/LinkIcon";
import { UnlinkIcon } from "./internal/UnlinkIcon";
import { NoLinkIcon } from "./internal/NoLinkIcon";
import { SearchIcon } from "./internal/SearchIcon";
import { ExportIcon } from "./internal/ExportIcon";
import { AddLineIcon } from "./internal/AddLineIcon";
import { MoreVertIcon } from "./internal/MoreVertIcon";
import { ArrowDoubleRightIcon } from "./internal/ArrowDoubleRightIcon";
import { DateRangeIcon } from "./internal/DateRangeIcon";
import { ImportIcon } from "./internal/ImportIcon";
import { ExternalLinkIcon } from "./internal/ExternalLinkIcon";
import { MobileIcon } from "./internal/MobileIcon";
import { DesktopIcon } from "./internal/DesktopIcon";
import { DesktopMobileIcon } from "./internal/DesktopMobileIcon";
import { CopyIcon } from "./internal/CopyIcon";
import { BarChartFramedIcon } from "./internal/BarChartFramedIcon";
import { LineChartFramedIcon } from "./internal/LineChartFramedIcon";
import { MultiLineChartFramedIcon } from "./internal/MultiLineChartFramedIcon";
import { TruckIcon } from "./internal/TruckIcon";
import { CameraMovieIcon } from "./internal/CameraMovieIcon";
import { CodeFileIcon } from "./internal/CodeFileIcon";
import { CartSecureIcon } from "./internal/CartSecureIcon";
import { ProfileIcon } from "./internal/ProfileIcon";
import { AuctionIcon } from "./internal/AuctionIcon";
import { MailIcon } from "./internal/MailIcon";
import { MailOpenIcon } from "./internal/MailOpenIcon";
import { FilterIcon } from "./internal/FilterIcon";
import { MoreIcon } from "./internal/MoreIcon";
import { NotificationIcon } from "./internal/NotificationIcon";
import { UserSettingsIcon } from "./internal/UserSettingsIcon";
import { CompanySettingsIcon } from "./internal/CompanySettingsIcon";
import { HeartPulseIcon } from "./internal/HeartPulseIcon";
import { CheckBoxCircleIcon } from "./internal/CheckBoxCircleIcon";
import { AlartIcon } from "./internal/AlartIcon";
import { BaseStationIcon } from "./internal/BaseStationIcon";
import { BrailleIcon } from "./internal/BrailleIcon";
import { LabelIcon } from "./internal/LabelIcon";
import { DownloadCloudIcon } from "./internal/DownloadCloudIcon";
import { SaveIcon } from "./internal/SaveIcon";
import { CloseCircleIcon } from "./internal/CloseCircleIcon";
import { BidStrapIcon } from "./internal/BidStrapIcon";
import { FluctIcon } from "./internal/FluctIcon";
import { DataStrapIcon } from "./internal/DataStrapIcon";

export type IconName =
  | "dashboard"
  | "bar_chart"
  | "bar_chart_framed"
  | "line_chart_framed"
  | "multi_line_chart_framed"
  | "logout"
  | "setting"
  | "arrow_bottom"
  | "arrow_left"
  | "arrow_right"
  | "pencil"
  | "delete_bin"
  | "close"
  | "forbid"
  | "check"
  | "eye"
  | "eye_off"
  | "exclamation"
  | "arrow_double_left"
  | "arrow_double_right"
  | "question"
  | "zoom_in"
  | "sort_up"
  | "sort_down"
  | "sort_inactive"
  | "folder"
  | "folder_open"
  | "return_line"
  | "link"
  | "unlink"
  | "no_link"
  | "search"
  | "import"
  | "export"
  | "add_line"
  | "more"
  | "more_vert"
  | "date_range"
  | "external_link"
  | "desktop"
  | "mobile"
  | "desktop_mobile"
  | "copy"
  | "truck"
  | "camera_movie"
  | "code_file"
  | "cart_secure"
  | "profile"
  | "auction"
  | "mail"
  | "mail_open"
  | "filter"
  | "notification"
  | "user_settings"
  | "company_settings"
  | "heart_pulse"
  | "checkbox_circle"
  | "alart"
  | "base_station"
  | "braille"
  | "label"
  | "download_cloud"
  | "save"
  | "close_circle"
  | "bid_strap"
  | "fluct"
  | "data_strap";

type IconType = "fill" | "line";
type IconColor = IconType | "active" | string;
type IconSize = "sm" | "md" | "lg";
export const iconSize: { [key in IconSize]: number } = {
  sm: 12,
  md: 18,
  lg: 24,
};
export type IconProps = {
  type: IconType;
  fill: string;
};

// for storybook
export const icons: {
  [key in IconName]: React.FunctionComponent<IconProps>;
} = {
  dashboard: DashboardIcon,
  bar_chart: BarChartIcon,
  bar_chart_framed: BarChartFramedIcon,
  line_chart_framed: LineChartFramedIcon,
  multi_line_chart_framed: MultiLineChartFramedIcon,
  logout: LogoutIcon,
  setting: SettingIcon,
  arrow_bottom: ArrowBottomIcon,
  arrow_left: ArrowLeftIcon,
  arrow_right: ArrowRightIcon,
  pencil: PencilIcon,
  delete_bin: DeleteBinIcon,
  close: CloseIcon,
  forbid: ForbidIcon,
  check: CheckIcon,
  eye: EyeIcon,
  eye_off: EyeOffIcon,
  exclamation: ExclamationIcon,
  arrow_double_left: ArrowDoubleLeftIcon,
  arrow_double_right: ArrowDoubleRightIcon,
  question: QuestionIcon,
  zoom_in: ZoomInIcon,
  sort_up: SortUpIcon,
  sort_down: SortDownIcon,
  sort_inactive: SortInactiveIcon,
  folder: FolderIcon,
  folder_open: FolderOpenIcon,
  return_line: ReturnLineIcon,
  link: LinkIcon,
  unlink: UnlinkIcon,
  no_link: NoLinkIcon,
  search: SearchIcon,
  import: ImportIcon,
  export: ExportIcon,
  add_line: AddLineIcon,
  more: MoreIcon,
  more_vert: MoreVertIcon,
  date_range: DateRangeIcon,
  external_link: ExternalLinkIcon,
  desktop: DesktopIcon,
  mobile: MobileIcon,
  desktop_mobile: DesktopMobileIcon,
  copy: CopyIcon,
  truck: TruckIcon,
  camera_movie: CameraMovieIcon,
  code_file: CodeFileIcon,
  cart_secure: CartSecureIcon,
  profile: ProfileIcon,
  auction: AuctionIcon,
  mail: MailIcon,
  mail_open: MailOpenIcon,
  filter: FilterIcon,
  notification: NotificationIcon,
  user_settings: UserSettingsIcon,
  company_settings: CompanySettingsIcon,
  heart_pulse: HeartPulseIcon,
  checkbox_circle: CheckBoxCircleIcon,
  alart: AlartIcon,
  base_station: BaseStationIcon,
  braille: BrailleIcon,
  label: LabelIcon,
  download_cloud: DownloadCloudIcon,
  save: SaveIcon,
  close_circle: CloseCircleIcon,
  bid_strap: BidStrapIcon,
  fluct: FluctIcon,
  data_strap: DataStrapIcon,
};

const iconFactory = (name: IconName) => (props: IconProps) => {
  const Component = icons[name];
  return <Component {...props} />;
};

const getIconColor = (color: IconColor, theme: Theme) => {
  switch (color) {
    case "fill":
    case "line":
    case "active":
      return theme.palette.icon[color];
    default:
      return color;
  }
};

export type Props = {
  name: IconName;
  type?: IconType;
  size?: IconSize;
  color?: IconColor;
};

const Icon = React.forwardRef<HTMLDivElement, Props>(
  ({ name, type = "line", size = "md", color = "fill" }, ref) => {
    const theme = useTheme();
    return (
      <Styled.Container ref={ref} size={iconSize[size]}>
        {iconFactory(name)({ type, fill: getIconColor(color, theme) })}
      </Styled.Container>
    );
  },
);

export default Icon;
