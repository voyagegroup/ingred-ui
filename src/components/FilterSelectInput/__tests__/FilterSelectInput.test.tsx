import * as React from "react";
import "@testing-library/jest-dom";
import { cleanup } from "@testing-library/react";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";
import { FilterSelectInput } from "../";
import Icon from "../../Icon";

describe("FileUploader component testing", () => {
  afterEach(cleanup);

  test("FileUploader", () => {
    const { asFragment } = renderWithThemeProvider(
      <FilterSelectInput
        value="項目1"
        values={[
          "項目1",
          "value2",
          "すごく長い値すごく長い値すごく長い値すごく長い値すごく長い値",
        ]}
        selectedIndex={0}
        selectOptions={[
          {
            icon: (
              <Icon name="operator_match" type="line" color="currentColor" />
            ),
            label: "含む",
          },
          {
            icon: (
              <Icon
                name="operator_does_not_match"
                type="line"
                color="currentColor"
              />
            ),
            label: "含まない",
          },
          {
            icon: (
              <Icon name="operator_contains" type="line" color="currentColor" />
            ),
            label: "いずれかを含む",
          },
        ]}
        onChange={jest.fn}
        onSelectChange={jest.fn}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
