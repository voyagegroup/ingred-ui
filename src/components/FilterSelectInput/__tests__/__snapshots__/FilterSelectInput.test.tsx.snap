// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUploader component testing FileUploader 1`] = `
<DocumentFragment>
  <div
    class="sc-hHLeRK cXDfAp"
    data-size="medium"
    data-small="false"
  >
    <button
      aria-expanded="false"
      aria-haspopup="dialog"
      aria-label="フィルターのタイプを選ぶ"
      class="sc-dmRaPn iRdULn"
      type="button"
    >
      <span
        class="sc-bczRLJ gJuzRf"
        size="18"
      >
        <svg
          viewBox="0 0 24 24"
        >
          <path
            d="M3.9 19.1V4.9h16.2v14.2zM2 4c0-.6.4-1 1-1h18c.6 0 1 .4 1 1v16c0 .6-.4 1-1 1H3a1 1 0 0 1-1-1zm11.1 9h-2.3L12 9.7zm0-5.5H11l-3.4 9h2l.7-2h3.4l.6 2h2.1z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <span
        class="sc-bczRLJ gJuzRf"
        size="18"
      >
        <svg
          viewBox="0 0 20 20"
        >
          <path
            d="m10 10.9 4.1-4.1 1.2 1.1-5.3 5.3L4.7 8l1.2-1.1z"
            fill="currentColor"
          />
        </svg>
      </span>
    </button>
    <div
      class="sc-iIPllB jDKTvv"
    >
      <button
        aria-expanded="false"
        aria-haspopup="dialog"
        class="sc-gicCDI iwSTkl"
        type="button"
      >
        <span
          class="sc-ezWOiH loKWCS"
        >
          項目1
        </span>
        <span
          class="sc-bZkfAO fZvvlw"
        >
          <span
            class="sc-bczRLJ gJuzRf"
            size="18"
          >
            <svg
              viewBox="0 0 20 20"
            >
              <path
                d="m10 10.9 4.1-4.1 1.2 1.1-5.3 5.3L4.7 8l1.2-1.1z"
                fill="currentColor"
              />
            </svg>
          </span>
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
